diff --git a/cmd/gost/main.go b/cmd/gost/main.go
index d3a5a42..4ea35a1 100644
--- a/cmd/gost/main.go
+++ b/cmd/gost/main.go
@@ -13,6 +13,7 @@
 	"sync"
 
 	"github.com/go-gost/core/logger"
+	helper "github.com/go-gost/gost/gost_helper"
 	xlogger "github.com/go-gost/x/logger"
 	"github.com/judwhite/go-svc"
 )
@@ -85,6 +86,9 @@
 func init() {
 	var printVersion bool
 
+	helper.Version = version
+	helper.PreInit()
+
 	flag.Var(&services, "L", "service list")
 	flag.Var(&nodes, "F", "chain node list")
 	flag.StringVar(&cfgFile, "C", "", "configuration file")
@@ -102,6 +106,8 @@
 		os.Exit(0)
 	}
+	helper.PostInit()
+
 }
 
 func main() {
 	log := xlogger.NewLogger()
